### ISSUE TITLE
Using interactor in Integration tests

### ISSUE THREAD

**Issue by benoror** [adopted-ember-addons/ember-pikaday#127]
Using interactor in Integration tests results in this error:

```
Cannot read property 'dispatchEvent' of undefined
```

This is the line in question: https://github.com/edgycircle/ember-pikaday/blob/fb16c12891179891d410ffff75c039bcaa9f37ea/addon/helpers/pikaday.js#L48

Could this be caused due to (`Ember.`)`$` selector working at the component's scope level, in integration tests?

**Comment by duizendnegen** [comment]
What version are you on? Would it be possible for you to cross-check between Ember versions?

**Comment by benoror** [comment]
I am using the latest version `2.2.0`, it fails in all ember-try scenarios. I pushed a branch so you can see the errors in Travis CI:

https://travis-ci.org/benoror/ember-formly/builds/196219423

**Comment by benoror** [comment]
the failing test: https://github.com/benoror/ember-formly/commit/4dc2ee659c15789488c57d444c4f954bc78e070f#diff-7ebef8b878a9e72976f90d3ee4b7ef9fR33

**Comment by duizendnegen** [comment]
Thanks for the response - yep, definitely a bug. I've only been using these interactor in acceptance tests, but from and end-user perspective, it should work fine in integration tests as well.

I'll try and find some cycles to investigate, but don't let that stop you from digging in yourself instead :)

**Comment by benoror** [comment]
Firstly I thought the issue was the `jquery` object (`$`) being scoped to the component's div, as I had the same issue with `ember-wormhole` & `ember-tether` stuff, where it couldn't find the DOM element in an injected outer div such as a modal, or a calendar in this case. 

But then I paused on the caught exception and found that the calendar day's `td`'s are not rendered by the time the `$` selector is ran:

![](http://g.recordit.co/sua72euPvN.gif)

**Comment by duizendnegen** [comment]
That's quite insightful - thanks for the pointer, appreciate it. I know my
acceptance test is sprinkled with andThen, separating the opening of the
picker and the clicking of an element

**Comment by benoror** [comment]
Interesting, I just tried wrapping the `selectDate` and the `assert` into a [`wait` helper](https://guides.emberjs.com/v2.7.0/testing/testing-components/#toc_waiting-on-asynchronous-behavior), which is similar to `andThen` but for integration tests, but got the very same error

```javascript
  let interactor = openDatepicker(this.$('input'));

  return wait().then(() => {
    interactor.selectDate(new Date(1989, 3, 28));
    assert.equal(this.get('model.test'), '28-03-1989', 'User interaction mutates the model');
  });
```

**Comment by duizendnegen** [comment]
As a side-note, if you're not super dependant on Pikaday styling and interactions, https://github.com/cibernox/ember-power-calendar could be a good alternative (also looking at the goals of your project) :)

**Comment by sevab** [comment]
Hi @benoror, getting same error in my integration tests.
Did you manage to find a workaround?

**Comment by benoror** [comment]
@sevab nope, still pending to test: https://github.com/benoror/ember-formly/issues/8

I haven't upgraded it tho, is it still an issue in newer versions?

**Comment by sevab** [comment]
@benoror, yep getting same issue in `v2.2.4`. But will probably migrate to `ember-power-calendar` as advised.

**Comment by alexlafroscia** [comment]
Can those having issues with the test helper check out the "new" (refactored) test helpers made available in `2.3.0`? They should not have issues with locating elements outside of the test container, and are build to be used with `async/await` to avoid issues around elements not being available before the picker is open.

**Comment by alexlafroscia** [comment]
I'm going to close this issue for now -- we're using the new interactor in the test suite for this addon without issues right now.

If anyone is having problems with the new test helpers, please make a new issue so we can get that fixed!
